relay-server/src/utils/envelope_context.rs

```diff
@@ -119,6 +125,19 @@ impl EnvelopeContext {
         }
     }
 
+    /// Returns the data category of the event item in the envelope.
+    ///
+    /// If metrics have been extracted from the event item, this will return the indexing category.
+    /// Outcomes for metrics (the base data category) will be logged by the metrics aggregator.
+    fn event_category(&self) -> Option<DataCategory> {
+        let category = self.summary.event_category?;
+
+        match category.index_category() {
+            Some(category) if self.summary.event_metrics_extracted => Some(category),
```

Comment:
I understand why @olksdr was confused, maybe this would help?

```suggestion
            Some(index_category) if self.summary.event_metrics_extracted => Some(index_category),
```